mypy/semanal.py

```diff
@@ -995,6 +994,53 @@ def analyze_property_with_multi_part_definition(self, defn: OverloadedFuncDef) -
+                                if not item.func.type:
+                                    item.func.type = callable_type(
+                                        item.func, self.named_type("builtins.function"), NoneType()
+                                    )
+                                    if len(item.func.type.arg_types) == 2:
+                                        item.func.type.arg_types[1] = first_item.func.type.ret_type
+                                elif isinstance(item.func.type, CallableType):
+                                    if (
+                                        len(item.func.type.arg_types) == 2
+                                        and is_unannotated_any(item.func.type.arg_types[1])
+                                    ):
+                                        item.func.type.arg_types[1] = first_item.func.type.ret_type
+                                    if is_unannotated_any(item.func.type.ret_type):
+                                        item.func.type.ret_type = NoneType()
+                            else:
+                                # set some boring defaults
+                                if not item.func.type:
+                                    item.func.type = callable_type(
+                                        item.func, self.named_type("builtins.function"),
+                                        UntypedType(TypeOfAny.explicit)
+                                    )
+                                    item.func.type.arg_types[1] = UntypedType(TypeOfAny.explicit)
+                                    item.func.type.implicit = True
+                                elif isinstance(item.func.type, CallableType):
+                                    if (
+                                        len(item.func.type.arg_types) == 2
+                                        and is_unannotated_any(item.func.type.arg_types[1])
+                                    ):
+                                        item.func.type.arg_types[1] = NoneType()
+                                    if is_unannotated_any(item.func.type.ret_type):
+                                        item.func.type.ret_type = NoneType()
+                        if node.name == 'deleter':
+                            # infer types from the getter
+                            if first_item.func.type:
+                                if not item.type:
+                                    item.func.type = callable_type(
+                                        item.func, self.named_type("builtins.function"), NoneType()
+                                    )
+                            else:
+                                if not item.func.type:
+                                    item.func.type = callable_type(
+                                        item.func, self.named_type("builtins.function"),
+                                        UntypedType(TypeOfAny.explicit)
+                                    )
```

Comment:
some of these are repeated, can they be variables
```suggestion
                            idk = callable_type(
                                item.func, self.named_type("builtins.function"), NoneType()
                            )
                            idk2 = callable_type(
                                item.func, self.named_type("builtins.function"),
                                UntypedType(TypeOfAny.explicit)
                            )
                            if isinstance(first_item.func.type, CallableType):
                                if not item.func.type:
                                    item.func.type = idk
                                    if len(item.func.type.arg_types) == 2:
                                        item.func.type.arg_types[1] = first_item.func.type.ret_type
                                elif isinstance(item.func.type, CallableType):
                                    if (
                                        len(item.func.type.arg_types) == 2
                                        and is_unannotated_any(item.func.type.arg_types[1])
                                    ):
                                        item.func.type.arg_types[1] = first_item.func.type.ret_type
                                    if is_unannotated_any(item.func.type.ret_type):
                                        item.func.type.ret_type = NoneType()
                            else:
                                # set some boring defaults
                                if not item.func.type:
                                    item.func.type = idk2
                                    item.func.type.arg_types[1] = UntypedType(TypeOfAny.explicit)
                                    item.func.type.implicit = True
                                elif isinstance(item.func.type, CallableType):
                                    if (
                                        len(item.func.type.arg_types) == 2
                                        and is_unannotated_any(item.func.type.arg_types[1])
                                    ):
                                        item.func.type.arg_types[1] = NoneType()
                                    if is_unannotated_any(item.func.type.ret_type):
                                        item.func.type.ret_type = NoneType()
                        if node.name == 'deleter':
                            # infer types from the getter
                            if first_item.func.type:
                                if not item.type:
                                    item.func.type = idk
                            else:
                                if not item.func.type:
                                    item.func.type = idk2
```

Reply:
Good pickup